doc/source/whatsnew/v1.2.0.rst

```diff
@@ -331,6 +331,7 @@ Plotting
 
 - Bug in :meth:`DataFrame.plot` was rotating xticklabels when ``subplots=True``, even if the x-axis wasn't an irregular time series (:issue:`29460`)
 - Bug in :meth:`DataFrame.plot` where a marker letter in the ``style`` keyword sometimes causes a ``ValueError`` (:issue:`21003`)
+- Twinned axes were losing their tick labels which should only happen to all but the last row or column of 'externally' shared axes (:issue:`33767`)
```

Comment:
this should be issue number not PR number, could you pls change to 33819 instead?

Reply:
Sure, np.